build-system/amp.extern.js

```diff
@@ -144,6 +144,9 @@ process.env.SERVE_MODE;
 window.IS_AMP_ALT;
 
 // Exposed to ads.
+// TODO: @erwinmombay I do believe this is safe to remove because
+// All are passed to ampcontext in JsonObject. While isMaster and master is
+// calculated in ampcontext-integration within the iframe.
```

Comment:
you'd have to change all the reads to use bracket notation if you want to remove this

Reply:
Here https://github.com/ampproject/amphtml/blob/master/3p/ampcontext.js#L294?

Reply:
Asked @erwinmombay offline. We want to leave the extern for future optimization that is not related to singlepass. I removed my comment. We will need to add missing variables to the externs, or use bracket notation.